services/beaconchain/src/main/java/tech/pegasys/teku/services/beaconchain/BeaconChainController.java

```diff
@@ -506,7 +507,7 @@ private void setupInitialState() {
 
   private void onStoreInitialized() {
     UnsignedLong genesisTime = recentChainData.getGenesisTime();
-    UnsignedLong currentTime = UnsignedLong.valueOf(System.currentTimeMillis() / 1000);
+    UnsignedLong currentTime = UnsignedLong.valueOf(Instant.now().getEpochSecond());
```

Comment:
Would be nice to avoid using `Instant.now()` here and get the time via `TimeProvider` if possible but if it's not available I can live with it since it was already using `System.currentTimeMillis`.  No real reason I can see to change this though.

Reply:
Switching to timeProvider.